### PR TITLE
Add support for managing jetty threads

### PR DESCRIPTION
**Pull request by bastelfreak**
This is based on https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/blob/master/doc/jetty-config.md

### CODE REVIEW

File: templates/server/puppetserver/conf.d/webserver.conf.erb [hunk 1]
```diff
@@ -27,4 +27,19 @@ webserver: {
         <%= cipher_suite %>,
     <%- end -%>
     ]
+<%- if @server_acceptor_threads -%>
```
Comment (ekohl) [inline templates/server/puppetserver/conf.d/webserver.conf.erb]: For them to be in the local scope you need to "import" them in the `puppet::server::puppetserver` class. We also "import" them in `puppet::server`.

**Comment by mmoll** [comment]
@bastelfreak this needs a rebase :)

**Comment by bastelfreak** [comment]
@mmoll @ekohl can you review this please? I think the code is fine, but something is fishy with the acceptance tests on centos 6.

**Comment by mmoll** [comment]
A test with these parameters set would be nice, looks fine visually.

**Comment by bastelfreak** [comment]
I'm not sure yet why the new tests fail. I will have a look again later. Maybe somebody else has an idea.

File: spec/classes/puppet_server_config_spec.rb [hunk 2]
```diff
@@ -714,6 +714,53 @@
         end
       end
 
+      describe 'alow jetty specific server threads' do
```
Comment (ekohl) [inline spec/classes/puppet_server_config_spec.rb]: Could you move these tests to `spec/classes/puppet_server_puppetserver_spec.rb` where we have all the tests just for puppetserver?

File: manifests/init.pp [hunk 3]
```diff
@@ -536,6 +536,32 @@
 #
 # $server_parser::                          Sets the parser to use. Valid options are 'current' or 'future'.
 #                                           Defaults to 'current'.
```
Comment (mmoll) [inline manifests/init.pp]: one comment line missing :point_up:

**Comment by mmoll** [comment]
@bastelfreak one tiny thing, otherwise IMHO ready to go :)

File: manifests/init.pp (hunk 3 above)
Comment (bastelfreak) [reply manifests/init.pp]: fixed

File: spec/classes/puppet_server_puppetserver_spec.rb [hunk 4]
```diff
@@ -586,6 +586,53 @@
         let(:params) { super().merge(server_puppetserver_version: '2.1.0') }
         it { should raise_error(Puppet::Error, /puppetserver <2.2 is not supported by this module version/) }
       end
+
+      describe 'allow jetty specific server threads' do
+        context 'with thread config' do
+          let :params do
+            {
+              server:                      true,
```
Comment (ekohl) [inline spec/classes/puppet_server_puppetserver_spec.rb]: Could you inherit from the parent?
```ruby
let(:params) do
  super().merge(
    server_selector_threads:     1,
    server_acceptor_threads:     2,
    server_ssl_selector_threads: 3,
    server_ssl_acceptor_threads: 4,
    server_max_threads:          5 
  )
end
```

File: spec/classes/puppet_server_puppetserver_spec.rb [hunk 5]
```diff
@@ -586,6 +586,53 @@
...
+            {
+              server:                  true,
+              server_implementation:   'puppetserver',
+              server_ca:               true,
+              server_puppetserver_dir: '/etc/custom/puppetserver',
+              server_jruby_gem_home:   '/opt/puppetlabs/server/data/puppetserver/jruby-gems'
+            }
+          end
+
+          it { is_expected.to compile.with_all_deps }
```
Comment (ekohl) [inline spec/classes/puppet_server_puppetserver_spec.rb]: Could you move this to the `with default parameters` block (starts at line 29)? Perhaps merge it in the existing check for this file at line 73.

File: spec/classes/puppet_server_puppetserver_spec.rb (hunk 4 above)
Comment (bastelfreak) [reply spec/classes/puppet_server_puppetserver_spec.rb]: done

File: spec/classes/puppet_server_puppetserver_spec.rb (hunk 5 above)
Comment (bastelfreak) [reply spec/classes/puppet_server_puppetserver_spec.rb]: done

**Comment by ekohl** [comment]
Thanks!

**Pull request merged**